### PR TITLE
Use `CoherenceModel` for `LdaModel.top_topics`

### LINKED ISSUE #1128: Use CoherenceModel in top_topics in LDAModel to avoid code duplication
'u_mass' coherence is currently implemented in two places in gensim: CoherenceModel and LDAModel.

Use `CoherenceModel(coherence='u_mass', topn=num_words)` in LDAModel [`top_topics`](https://github.com/RaRe-Technologies/gensim/blob/develop/gensim/models/ldamodel.py#L824)

Will require changes to CoherenceModel to return non-aggregated coherence scores, topic words and avoid importing LDAModel.

### PR DESCRIPTION
**Pull request by macks22**
This will resolve #1128.

### CODE REVIEW

**Comment by menshikh-iv** [comment]
@chinmayapancholi13 this PR fix problem with coherence and SklWrappers?

File: gensim/models/ldamodel.py [hunk 1]
```diff
@@ -823,72 +830,32 @@ def get_topic_terms(self, topicid, topn=10):
...
         """
-        topic = self.state.get_lambda()[topicid]
+        topic = self.get_topics()[topicid]
         topic = topic / topic.sum()  # normalize to probability distribution
         bestn = matutils.argsort(topic, topn, reverse=True)
         return [(id, topic[id]) for id in bestn]
 
-    def top_topics(self, corpus, num_words=20):
+    def top_topics(self, corpus=None, texts=None, dictionary=None, window_size=None,
+                   coherence='u_mass', topn=20, processes=-1):
```
Comment (menshikh-iv) [inline gensim/models/ldamodel.py]: I don't think that we should use all coherence types for this method, `u_mass` will be enough for it (for this reason please remove coherence parameters from arguments)

File: gensim/models/ldamodel.py [hunk 2]
```diff
@@ -823,72 +830,32 @@ def get_topic_terms(self, topicid, topn=10):
...
         """
-        Calculate the Umass topic coherence for each topic. Algorithm from
-        **Mimno, Wallach, Talley, Leenders, McCallum: Optimizing Semantic Coherence in Topic Models, CEMNLP 2011.**
+        Calculate the coherence for each topic; default is Umass coherence. See the
```
Comment (menshikh-iv) [inline gensim/models/ldamodel.py]: Googe-style docstring (here and everywhere)

File: gensim/models/ldamodel.py (hunk 1 above)
Comment (macks22) [inline gensim/models/ldamodel.py]: Several papers have shown that sliding-window based coherence metrics have higher correlation with human judgements, and `u_mass` has lower correlations with human judgements than these methods. Why limit to `u_mass` when it is not the best technique for coherence calculations?

File: gensim/models/ldamodel.py (hunk 2 above)
Comment (macks22) [inline gensim/models/ldamodel.py]: Not sure what you're looking for here, though I've noticed this comment in several of my PRs. To clarify, are you asking for the beginning line to start on the same line as the start quotes (""")? Or are you asking that I include the ``Args`` specification for this method and the others? Thanks!

**Comment by macks22** [comment]
@menshikh-iv when you have a moment, I'd love your thoughts on my comments. Based on your comments, I think this may still need some changes, but I need your guidance on how to make them. Thanks!

File: gensim/models/ldamodel.py (hunk 1 above)
Comment (menshikh-iv) [inline gensim/models/ldamodel.py]: I agree with you, but for "sliding windows" approach user should pass `corpus`, `texts` and `dictionary`, it's slightly complicated.

Although perhaps you are right, if the default method is `UMass` I see no difference for users.

File: gensim/models/ldamodel.py (hunk 2 above)
Comment (menshikh-iv) [inline gensim/models/ldamodel.py]: Sorry for the confusion, I mean `Args` and `Returns` sections.

**Comment by menshikh-iv** [comment]
Sorry for waiting, please fix doc-strings and I'll merge this PR :+1:

File: gensim/models/ldamodel.py (hunk 2 above)
Comment (macks22) [inline gensim/models/ldamodel.py]: fixed throughout

**Comment by macks22** [comment]
@menshikh-iv I'm not sure how this error relates to my PR; maybe something with bad scikit version in build server? The current build is only failing due to this error in `test_sklearn_api`:
```
======================================================================
ERROR: testPipeline (gensim.test.test_sklearn_api.TestLdaSeqWrapper)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/RaRe-Technologies/gensim/gensim/test/test_sklearn_api.py", line 340, in testPipeline
    text_ldaseq.fit(corpus, test_target)
  File "/home/travis/miniconda2/envs/gensim-test/lib/python3.5/site-packages/sklearn/pipeline.py", line 257, in fit
    Xt, fit_params = self._fit(X, y, **fit_params)
  File "/home/travis/miniconda2/envs/gensim-test/lib/python3.5/site-packages/sklearn/pipeline.py", line 226, in _fit
    self.steps[step_idx] = (name, fitted_transformer)
TypeError: 'tuple' object does not support item assignment
```

**Comment by menshikh-iv** [comment]
@macks22 this problem isn't related to your changes. This problem connected with new sklearn release, we will fix it soon.

**Comment by menshikh-iv** [comment]
Nice work @macks22, you are very productive :+1:

**Pull request merged**